src/transformers/models/wav2vec2_conformer/modeling_wav2vec2_conformer.py

```diff
@@ -884,6 +888,8 @@ def forward(
                 attention_mask.shape[0], 1, attention_mask.shape[-1], attention_mask.shape[-1]
             )
 
+        if not self.post_layer_norm:
```

Comment:
We add this new post layer-norm logic in the modelling code, but not in the conversion script. Is it actually used by any checkpoints?

Reply:
@sanchit-gandhi 
I didn't exactly understand the intent of the question, but it also applies to checkpoints that use `layer_norm_first` as False in the fairseq code.

https://github.com/facebookresearch/fairseq/blob/main/fairseq/models/wav2vec/wav2vec2.py#L1237-L1238

Reply:
Apologies! Let me clarify :) I notice here that we've:
1. Added the `post_layer_norm` logic to the modelling code
2. Have not added the `post_layer_norm` option in the conversion script

Is the `post_layer_norm` logic actually used by any checkpoints we wish to convert? Note that `fairseq` might have extra functionality that isn't actually used in published a models. In this case, we tend not to add this functionality to Transformers, in order to keep the codebase as simple as possible.

In this case, if `post_layer_norm=False` is never actually used by any official checkpoints, it would be cleaner to omit this argument from the modelling code and config.

Reply:
Let me know if the above line of reasoning makes sense @upskyy!

Reply:
Bumping this thread - would be interested in hearing your thoughts @upskyy!